internal/pkg/cli/svc_deploy.go

```diff
@@ -424,62 +431,76 @@ func (o *deploySvcOpts) configureContainerImage() error {
 }
 
 func (o *deploySvcOpts) dfBuildArgs(svc interface{}) (*dockerengine.BuildArguments, error) {
-	copilotDir, err := o.ws.CopilotDirPath()
+	if err := o.retrieveWorkspacePath(); err != nil {
+		return nil, err
+	}
+	return buildArgs(o.name, o.imageTag, o.workspacePath, svc)
+}
+
+func (o *deploySvcOpts) pushArtifactsToS3() error {
+	mft, err := o.manifest()
 	if err != nil {
-		return nil, fmt.Errorf("get copilot directory: %w", err)
+		return err
 	}
-	return buildArgs(o.name, o.imageTag, copilotDir, svc)
+	if err := o.pushEnvFilesToS3Bucket(envFile(mft)); err != nil {
+		return err
+	}
+	return o.pushAddonsTemplateToS3Bucket()
 }
 
-func buildArgs(name, imageTag, copilotDir string, unmarshaledManifest interface{}) (*dockerengine.BuildArguments, error) {
-	type dfArgs interface {
-		BuildArgs(rootDirectory string) *manifest.DockerBuildArgs
-		ContainerPlatform() string
+func (o *deploySvcOpts) pushEnvFilesToS3Bucket(path string) error {
+	if path == "" {
+		return nil
 	}
-	mf, ok := unmarshaledManifest.(dfArgs)
-	if !ok {
-		return nil, fmt.Errorf("%s does not have required methods BuildArgs() and ContainerPlatform()", name)
+	if err := o.retrieveWorkspacePath(); err != nil {
+		return err
 	}
-	var tags []string
-	if imageTag != "" {
-		tags = append(tags, imageTag)
+	content, err := o.fs.ReadFile(filepath.Join(o.workspacePath, path))
+	if err != nil {
+		return fmt.Errorf("read env file %s: %w", path, err)
 	}
```

Comment:
What will the guidance for the env file look like? I'm worried that users aren't aware of buildArgs that are fetched under the hood (like `platform`), and won't know they have to put them in their env file.

Reply:
It'll be the same as the dockerfile which we'll look in their workspace. I'll try to make that clear in the docs.

> won't know they have to put them in their env file.

What does this mean 🤔

Reply:
Do you have a link to the issue? I've been looking but can't find it. Clarifying the ask will help me! 🙏🏼 😸

Reply:
https://github.com/aws/copilot-cli/issues/1366 ?

Reply:
Ah, thank you. NVM ☝🏼 . I get it now.